src/Libraries/Microsoft.Extensions.Caching.Hybrid/Internal/DefaultJsonSerializerFactory.cs

```diff
@@ -1,42 +1,79 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
 using System.Buffers;
 using System.Diagnostics.CodeAnalysis;
 using System.Text.Json;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.Extensions.Caching.Hybrid.Internal;
 
 internal sealed class DefaultJsonSerializerFactory : IHybridCacheSerializerFactory
 {
+    private readonly IServiceProvider _serviceProvider;
+
+    internal static JsonSerializerOptions FieldEnabledJsonOptions { get; } = new() { IncludeFields = true };
+
+    internal JsonSerializerOptions Options { get; }
+
+    public DefaultJsonSerializerFactory(IServiceProvider serviceProvider)
+    {
+        // store the service provider and obtain the default JSON options, keyed by the **open** generic interface type
+        _serviceProvider = serviceProvider;
+
+#pragma warning disable IDE0079 // unnecessary suppression: TFM-dependent
+#pragma warning disable IL2026, IL3050 // AOT bits
+        Options = serviceProvider.GetKeyedService<JsonSerializerOptions>(typeof(IHybridCacheSerializer<>)) ?? JsonSerializerOptions.Default;
+#pragma warning restore IL2026, IL3050
+#pragma warning restore IDE0079
+    }
+
     public bool TryCreateSerializer<T>([NotNullWhen(true)] out IHybridCacheSerializer<T>? serializer)
     {
-        // no restriction
-        serializer = new DefaultJsonSerializer<T>();
+        // no restriction - accept any type (i.e. always return true)
+
+        // see if there is a per-type options registered (keyed by the **closed** generic type), otherwise use the default
+        JsonSerializerOptions options = _serviceProvider.GetKeyedService<JsonSerializerOptions>(typeof(IHybridCacheSerializer<T>)) ?? Options;
```

Comment:
why is resolving every time required? Did not we do it in ctor already? Or the idea is that if it was not yet available in ctor, but when performing `TryCreateSerializer<T>` it will be?

Reply:
It isn't; this method is only called once per `T`, with the selected serializer then cached in `DefaultHybridCache` - but we don't know which various `T` we will need until we see them at runtime.